src/dmd/backend/debugprint.d

```diff
@@ -0,0 +1,486 @@
+        printf("\tBpred: "); WRblocklist(b.Bpred);
+        printf("\tBsucc: "); WRblocklist(b.Bsucc);
+        if (b.Belem)
+        {       if (debugf)                     /* if full output       */
+                        elem_print(b.Belem);
+                else
+                {       ferr("\t");
+                        WReqn(b.Belem);
+                        printf(";\n");
+                }
+        }
+        version (MARS)
+        {
+        if (b.Bcode)
+            b.Bcode.print();
+        }
+        version (SCPP)
+        {
+        if (b.Bcode)
+            b.Bcode.print();
+        }
+        ferr("\n");
+    }
+    else
+    {
+        targ_llong *pu;
+        int ncases;
+
+        assert(b);
+        printf("%2d: ", b.Bnumber); WRBC(b.BC);
+        if (b.Btry)
+            printf(" Btry=B%d",b.Btry ? b.Btry.Bnumber : 0);
+        if (b.Bindex)
+            printf(" Bindex=%d",b.Bindex);
+        if (b.BC == BC_finally)
+            printf(" b_ret=B%d", b.b_ret ? b.b_ret.Bnumber : 0);
+version (MARS)
+{
+        if (b.Bsrcpos.Sfilename)
+            printf(" %s(%u)", b.Bsrcpos.Sfilename, b.Bsrcpos.Slinnum);
+}
+        printf("\n");
+        if (b.Belem)
+        {
+            if (debugf)
+                elem_print(b.Belem);
+            else
+            {
+                ferr("\t");
+                WReqn(b.Belem);
+                printf(";\n");
+            }
+        }
+        if (b.Bpred)
+        {
+            printf("\tBpred:");
+            for (list_t bl = b.Bpred; bl; bl = list_next(bl))
+                printf(" B%d",list_block(bl).Bnumber);
+            printf("\n");
+        }
+        list_t bl = b.Bsucc;
+        switch (b.BC)
+        {
+            case BCswitch:
+                pu = b.Bswitch;
+                assert(pu);
+                ncases = cast(int)*pu;
+                printf("\tncases = %d\n",ncases);
+                printf("\tdefault: B%d\n",list_block(bl) ? list_block(bl).Bnumber : 0);
+                while (ncases--)
+                {   bl = list_next(bl);
+                    printf("\tcase %lld: B%d\n", cast(long)*++pu,list_block(bl).Bnumber);
+                }
+                break;
+            case BCiftrue:
+            case BCgoto:
+            case BCasm:
+            case BCtry:
+            case BCcatch:
+            case BCjcatch:
+            case BC_try:
+            case BC_filter:
+            case BC_finally:
+            case BC_lpad:
+            case BC_ret:
+            case BC_except:
+
+            Lsucc:
+                if (bl)
+                {
+                    printf("\tBsucc:");
+                    for ( ; bl; bl = list_next(bl))
+                        printf(" B%d",list_block(bl).Bnumber);
+                    printf("\n");
+                }
+                break;
+            case BCret:
+            case BCretexp:
+            case BCexit:
+                break;
+            default:
+                printf("bc = %d\n", b.BC);
```

Comment:
First rule of conversion: this isn't part of the CXX version.

Reply:
same as other comment.